# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/chromium/src/chrome/common/ipc_message.cc
# Commit: b09ae4c3a94b
# Full Hash: b09ae4c3a94b663279189c50172c1f403d499183
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2021-06-22 21:29:07
# Regressor Bug: 1706374
# File Overlap Count: 1
# Description:
#   Bug 1706374 - Part 11: Add NodeController component bridging IPC and Ports, r=handyman
#   
#   The NodeController and NodeChannel types act as the backbone connecting the
#   existing IPC logic and driving the ports routing code. Individual NodeChannel
#   objects wrap and respond to messages from IPC::Channel, and the NodeController
# ==============================================================================

diff -r 04422175004b -r b09ae4c3a94b ipc/chromium/src/chrome/common/ipc_message.cc
--- a/ipc/chromium/src/chrome/common/ipc_message.cc	Mon Jun 21 21:53:10 2021 +0000
+++ b/ipc/chromium/src/chrome/common/ipc_message.cc	Mon Jun 21 21:53:10 2021 +0000
@@ -21,9 +21,12 @@
 
 //------------------------------------------------------------------------------
 
+const mojo::core::ports::UserMessage::TypeInfo Message::kUserMessageTypeInfo{};
+
 Message::~Message() { MOZ_COUNT_DTOR(IPC::Message); }
 
-Message::Message() : Pickle(sizeof(Header)) {
+Message::Message()
+    : UserMessage(&kUserMessageTypeInfo), Pickle(sizeof(Header)) {
   MOZ_COUNT_CTOR(IPC::Message);
   header()->routing = header()->type = 0;
 #if defined(OS_POSIX)
@@ -33,7 +36,8 @@
 
 Message::Message(int32_t routing_id, msgid_t type, uint32_t segment_capacity,
                  HeaderFlags flags, bool recordWriteLatency)
-    : Pickle(sizeof(Header), segment_capacity) {
+    : UserMessage(&kUserMessageTypeInfo),
+      Pickle(sizeof(Header), segment_capacity) {
   MOZ_COUNT_CTOR(IPC::Message);
   header()->routing = routing_id;
   header()->type = type;
@@ -54,11 +58,13 @@
 }
 
 Message::Message(const char* data, int data_len)
-    : Pickle(sizeof(Header), data, data_len) {
+    : UserMessage(&kUserMessageTypeInfo),
+      Pickle(sizeof(Header), data, data_len) {
   MOZ_COUNT_CTOR(IPC::Message);
 }
 
-Message::Message(Message&& other) : Pickle(std::move(other)) {
+Message::Message(Message&& other)
+    : UserMessage(&kUserMessageTypeInfo), Pickle(std::move(other)) {
   MOZ_COUNT_CTOR(IPC::Message);
 #if defined(OS_POSIX)
   file_descriptor_set_ = std::move(other.file_descriptor_set_);